crates/typed-store/src/rocks/mod.rs

```diff
@@ -194,23 +214,44 @@ impl<K, V> DBMap<K, V> {
 pub struct DBBatch {
     rocksdb: Arc<rocksdb::DBWithThreadMode<MultiThreaded>>,
     batch: WriteBatch,
+    db_metrics: Arc<DBMetrics>,
 }
 
 impl DBBatch {
     /// Create a new batch associated with a DB reference.
     ///
     /// Use `open_cf` to get the DB reference or an existing open database.
-    pub fn new(dbref: &Arc<rocksdb::DBWithThreadMode<MultiThreaded>>) -> Self {
+    pub fn new(
+        dbref: &Arc<rocksdb::DBWithThreadMode<MultiThreaded>>,
+        db_metrics: &Arc<DBMetrics>,
+    ) -> Self {
         DBBatch {
             rocksdb: dbref.clone(),
             batch: WriteBatch::default(),
+            db_metrics: db_metrics.clone(),
         }
     }
 
     /// Consume the batch and write its operations to the database
     #[instrument(level = "trace", skip_all, err)]
     pub fn write(self) -> Result<(), TypedStoreError> {
+        let db_name = self
+            .rocksdb
+            .path()
+            .file_name()
+            .and_then(|f| f.to_str())
```

Comment:
I'm worried we are allocating here for every batch  write.

Reply:
ok i'll move this up to the struct